hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesState.java

```diff
@@ -273,9 +273,13 @@ public void init(
     @Override
     public void handleConsensusRound(final Round round, final SwirldDualState dualState) {
         throwIfImmutable();
+
         final var app = metadata.app();
+        app.mapWarmer().warmCache(round);
+
         app.dualStateAccessor().setDualState(dualState);
         app.logic().incorporateConsensus(round);
+        app.mapWarmer().cancelPendingWarmups(round.getRoundNum());
```

Comment:
One possibility is to call `cancelPendingWarmups` at the start of `warmCache(round)` with a simple implementation of just clearing the queue (no check for tasks' round numbers). That would reduce `EntityMapWarmer` exposure, simplify its implementation and reduce overhead.

Reply:
There was a `TODO: is this useful` in the code for cancelling based on round number before I submitted this PR. Sounds like it's not needed at all. I'll remove it.